src/ray/raylet/worker.cc

```diff
@@ -196,6 +198,15 @@ const BundleID &Worker::GetBundleId() const { return bundle_id_; }
 
 void Worker::SetBundleId(const BundleID &bundle_id) { bundle_id_ = bundle_id; }
 
+void Worker::SetJobId(const JobID &job_id) {
+  if (assigned_job_id_.IsNil()) {
+    assigned_job_id_ = job_id;
+  }
+
+  RAY_CHECK(assigned_job_id_ == job_id)
+      << "SetJobId failed: " << assigned_job_id_.Hex() << ":" << job_id.Hex();
```

Comment:
```suggestion
      << "SetJobId failed because it expected job_id to equal assigned job id: " << assigned_job_id_.Hex() << ":" << job_id.Hex();
```